ludwig/backend/ray.py

```diff
@@ -813,6 +817,45 @@ def initialize(self):
         # Disable placement groups on dask
         dask.config.set(annotations={"ray_remote_args": {"placement_group": None}})
 
+    def _update_dask_backend_with_pg(self, pg: PlacementGroup):
+        dask.config.set(annotations={"ray_remote_args": {"placement_group": pg}})
+
+    def _clear_dask_backend_pg(self):
+        dask.config.set(annotations={"ray_remote_args": {"placement_group": None}})
+
+    def provision_preprocessing_workers(self):
+        if not self._preprocessor_kwargs.get("use_preprocessing_placement_group", False):
+            logger.warning(
+                "Backend config has use_preprocessing_placement_group set to False or did not set it at all."
+                " provision_preprocessing_workers() is a no-op in this case."
+            )
+            return
+        num_cpu = self._preprocessor_kwargs["num_cpu_workers"]
+        self._preprocessor_pg = placement_group([{"CPU": num_cpu}])
+        ready = self._preprocessor_pg.wait(THREE_MINS_IN_S)
```

Comment:
The timeout makes me a little nervous. It could be the case that we're autoscaling up preprocessing resources, or that we're performing preprocessing as part of hyperopt and so resources may be available when a trial finishes.

Is it strictly necessary to raise an error here, or could we get away with just logging that we're currently waiting for resources?

I do like that the ValueError is pretty descriptive, so if we do run into this it should be easy enough to debug and possibly increase the timeout or remove it altogether if needed. What do you think?

Reply:
Since this is early days in the placement group work explicit failures might be helpful here. Also with ray trainer it fails if you try to provision resources and it hits a timeout so I would prefer to stay on the same set up as that